src/utils/index.test.ts

```diff
@@ -25,54 +26,209 @@ describe('Utils', () => {
+        '<iframe loading="auto" src="https://www.example.com/?id=props.id" width="150" height="100"></iframe>',
+      );
+      expect(result.scripts).toEqual(null);
+    });
+
+    it('should pass scripts and correctly assign params if available', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+          },
+        },
+        scripts: [
+          {
+            url: 'https://www.example.com',
+            params: ['id'],
+            strategy: 'worker',
+            location: 'head',
+            action: 'append',
+          },
+        ],
+      };
+
+      const inputs = {
+        id: 'userDefinedId',
+      };
+
+      const result = formatData(data as Data, inputs);
+      expect(result.html).toEqual('<iframe loading="lazy"></iframe>');
+      expect(result.scripts).not.toEqual(null);
+      expect(result.scripts!.length).toEqual(1);
+      expect(result.scripts![0].url).toEqual(
+        'https://www.example.com/?id=userDefinedId',
+      );
+    });
+
+    it('should forward all additional inputs as html attributes if not used elsewhere', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            width: '100',
+            height: '100',
+          },
+        },
+      };
+
+      const inputs = {
+        id: 'props.id',
+        loading: 'auto',
+        width: '150',
+      };
+
+      const result = formatData(data, inputs);
+      expect(result.html).toEqual(
+        '<iframe loading="auto" width="150" height="100" id="props.id"></iframe>',
+      );
+      expect(result.scripts).toEqual(null);
+    });
+
+    it('should include the user inputted slug to the src URL if provided as a parameter', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            src: {
+              url: 'https://www.example.com/',
+              slugParam: 'inputSlug',
+            },
+          },
+        },
+      };
+
+      const inputs = {
+        inputSlug: 'cool-slug',
+      };
+
+      const result = formatData(data, inputs);
+      expect(result.html).toEqual(
+        '<iframe loading="lazy" src="https://www.example.com/cool-slug"></iframe>',
+      );
+      expect(result.scripts).toEqual(null);
+    });
+
+    it('should replace the already existing slug if the user includes a slug parameter slug', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            src: {
+              url: 'https://www.google.com/maps/embed/v1/place',
```

Comment:
Oh, does this replace only the last part of the url?

Reply:
Yep!